modules/tinymce/src/themes/silver/main/ts/ui/dropdown/CommonDropdown.ts

```diff
@@ -158,8 +160,18 @@ const renderCommonDropdown = <T>(
 
       dropdownBehaviours: Behaviour.derive([
         ...spec.dropdownBehaviours,
-        DisablingConfigs.button(() => spec.disabled || sharedBackstage.providers.isDisabled()),
-        ReadOnly.receivingConfig(),
+        DisablingConfigs.button(() => {
+          if (prefix === MenuButtonClasses.Button) {
+            return sharedBackstage.providers.isDisabled() && !sharedBackstage.providers.isReadOnlyEnableUi();
+          }
```

Comment:
Could we simplify this a bit?

Perhaps pass some sort of "are we rendering ui" boolean into a new `sharedBackstage.providers` method that handles the "is ui enabled in readonly" flag itself? rather than constantly repeating the logic here?

Reply:
This has been changed as well, the backstage now checks the button allowedModes and against the current editor mode.